src/Stryker.Core/Stryker.Core/MutantFilters/BroadcastMutantFilter.cs

```diff
@@ -8,42 +8,33 @@ namespace Stryker.Core.MutantFilters
 {
     public class BroadcastMutantFilter : IMutantFilter
     {
+        public MutantFilter Type => MutantFilter.Broadcast;
         public IEnumerable<IMutantFilter> MutantFilters { get; }
 
-        public BroadcastMutantFilter(IEnumerable<IMutantFilter> mutantFilters)
-        {
-            MutantFilters = mutantFilters;
-        }
+        public BroadcastMutantFilter(IEnumerable<IMutantFilter> mutantFilters) => MutantFilters = mutantFilters;
 
         public string DisplayName => "broadcast filter";
 
         public IEnumerable<Mutant> FilterMutants(IEnumerable<Mutant> mutants, ReadOnlyFileLeaf file, StrykerOptions options)
         {
-            IEnumerable<Mutant> mutantsToTest = mutants.ToList();
+            IEnumerable<Mutant> mutantsToTest = mutants.Where(m => m.ResultStatus is not MutantStatus.Ignored).ToList();
 
-            // Then all other filters except diff filter
-            foreach (var mutantFilter in MutantFilters.Where(f => !(f is DiffMutantFilter)))
+            foreach (var mutantFilter in MutantFilters)
             {
                 // These mutants should be tested according to current filter
-                var extraMutantsToTest = mutantFilter.FilterMutants(mutantsToTest, file, options);
+                var remainingMutantsToTest = mutantFilter.FilterMutants(mutantsToTest, file, options);
 
                 // All mutants that weren't filtered out by a previous filter but were by the current filter are set to Ignored
-                foreach (var skippedMutant in mutantsToTest.Except(extraMutantsToTest))
+                foreach (var skippedMutant in mutantsToTest.Except(remainingMutantsToTest))
                 {
                     skippedMutant.ResultStatus = MutantStatus.Ignored;
                     skippedMutant.ResultStatusReason = $"Removed by {mutantFilter.DisplayName}";
                 }
 
-                mutantsToTest = extraMutantsToTest;
-            }
-
-            // Diff filter goes last if it is enabled
```

Comment:
This should never have been the job of the broadcast mutant filter, this is now solved by making the sorting of the mutant filters explicit when created in the mutant filter factory by using a sortedset.